lib/manager/components/transform/NormalizeField.js

```diff
@@ -0,0 +1,339 @@
+              Exceptions:
+              <FormGroup>
+                <Select.Creatable
+                  arrowRenderer={null}
+                  clearable={false}
+                  multi
+                  onChange={this._onChangeCapitalizeExceptions}
+                  placeholder='Enter exceptions then press enter'
+                  value={capitalizeExceptionList}
+                />
+              </FormGroup>
+            </span>
+          </li>
+          <li>
+            <Checkbox
+              checked={performSubstitutions}
+              onChange={this._onChangePerformSubstitutions}
+            >
+              Perform the following substitutions:
+            </Checkbox>
+            <FormGroup>
+              <Table condensed>
+                <thead>
+                  <tr>
+                    <th>Substitute</th>
+                    <th>With</th>
+                    <th>Normalize space?</th>
+                    <th />
+                  </tr>
+                </thead>
+                <tbody>
+                  {substitutions && substitutions.map((s, i) => (
+                    <SubstitutionRow
+                      index={i}
+                      key={i}
+                      onChange={this._onChangeSubstitution}
+                      onRemove={this._onRemoveSubstitution}
+                      substitution={s}
+                    />
+                  ))}
+                  <tr>
+                    <td colSpan='4'>
+                      <Button bsSize='xsmall' onClick={this._onNewSubstitution}>
+                        <Icon type='plus' /> Add
+                      </Button>
+                    </td>
+                  </tr>
+                </tbody>
+              </Table>
+            </FormGroup>
+          </li>
+        </ul>
+        <div style={{marginBottom: '10px'}}>
+          <Button
+            bsSize='xsmall'
+            disabled={inputIsUnchanged}
+            onClick={this._onSaveTransformationSettings}
+            style={{marginRight: '5px'}}
+          >
+            Save
+          </Button>
+          <Button
+            bsSize='xsmall'
+            disabled={inputIsUnchanged}
+            onClick={this._onRevertTransformationSettings}
+            style={{marginRight: '5px'}}
+          >
+            Revert
+          </Button>
+        </div>
+      </div>
+    )
+  }
+}
+
+type GtfsFieldSelectorProps = {
+  onChange: (ReactSelectOption => void),
+  selectedField: ?string,
+  tableName: ?string
+}
+
+/**
+ * Obtains the desired GTFS spec table.
+ */
+function getGtfsTableSpec (tableName: string): ?GtfsSpecTable {
+  const spec = getGtfsSpec()
+  if (!spec) {
+    throw new Error('GTFS spec could not be found!')
+  }
+  const tableFile = `${tableName}.txt`
+  return spec.find(t => t.name === tableFile)
+}
+
+/**
+ * Dropdown selector to select a field from a given GTFS table.
+ */
+const GtfsFieldSelector = (props: GtfsFieldSelectorProps) => {
```

Comment:
I could see this getting reused, so it would be nice to place it in its own file.

Reply:
Moved in https://github.com/ibi-group/datatools-ui/pull/649/commits/19fb3d2f7d2a415b941f3b47e667877c0958fe69.